### PR TITLE
feat: remove bucket creation from inside the module and refactor k3s/kind variant

### PR DESCRIPTION
**Pull request by lentidas**

### CODE REVIEW

**Comment by fsismondi** [comment]
Nice! can we merge this?

**Review by fsismondi — APPROVED** [review]

**Comment by modridi** [comment]
@lentidas Apart from some variables that I changed (see last 2 commits), this works fine on Azure. We can merge if the tests succeed on the remaining cloud providers.

**Review by modridi — APPROVED** [review]

File: aks/extra-variables.tf [hunk 1]
```diff
@@ -1,16 +1,8 @@
...
-variable "storage_account_replication_type" {
-  description = "Storage account replication type for storing loki logs"
-  default     = "LRS"
-  type        = string
-}
-
-variable "resource_group_name" {
-  description = "The Resource Group where the Managed Kubernetes Cluster should exist."
-  type        = string
+variable "logs_storage" {
```
Comment (fsismondi) [inline aks/extra-variables.tf:1]: this should be a sensitive var right?

Comment (lentidas) [reply aks/extra-variables.tf:1]: We need to deal with this latter, as per our discussion in our channel.

**Pull request merged**